src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SuddenHighwayTypeChangeCheck.java

```diff
@@ -0,0 +1,248 @@
+package org.openstreetmap.atlas.checks.validation.linear.edges;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.openstreetmap.atlas.checks.atlas.predicates.TypePredicates;
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.items.Node;
+import org.openstreetmap.atlas.geography.atlas.walker.OsmWayWalker;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.JunctionTag;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * Auto generated Check template
+ *
+ * @author v-garei
+ */
+public class SuddenHighwayTypeChangeCheck extends BaseCheck<Long>
+{
+    private static final String SUDDEN_HIGHWAY_TYPE_CHANGE_INSTRUCTION = "Way {0,number,#} has a connected edge which jumps significantly in highway classification. Please make sure the highway tag is not suspicious.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Collections
+            .singletonList(SUDDEN_HIGHWAY_TYPE_CHANGE_INSTRUCTION);
+    private static final String HIGHWAY_MINIMUM_DEFAULT = HighwayTag.RESIDENTIAL.toString();
+    private final HighwayTag minHighwayType;
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public SuddenHighwayTypeChangeCheck(final Configuration configuration)
+    {
+        super(configuration);
+        final String highwayType = this.configurationValue(configuration, "minHighwayType",
+                HIGHWAY_MINIMUM_DEFAULT);
+        this.minHighwayType = Enum.valueOf(HighwayTag.class, highwayType.toUpperCase());
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        if (TypePredicates.IS_EDGE.test(object) && ((Edge) object).isMainEdge()
+                && !isFlagged(object.getOsmIdentifier()))
+        {
+            final Edge edge = (Edge) object;
+            return HighwayTag.isCarNavigableHighway(edge) && HighwayTag.highwayTag(edge).isPresent()
```

Comment:
`HighwayTag.highwayTag(edge).isPresent()` would be redundant after `HighwayTag.isCarNavigableHighway(edge)`

Reply:
updated!